### PR TITLE
Report

### PR DESCRIPTION
**Pull request by jimshowalter**

### CODE REVIEW

File: src/app/services/words.service.ts [hunk 1]
```diff
@@ -35,20 +38,50 @@ export class WordsService implements OnDestroy {
...
+        map(words =>
+          words
+            .filter(x => x.learned === false)
+            .sort((word1, word2) => {
+              return this.sort(word1, word2, criteria.direction, criteria.sortColumn);
+            })
+            .slice(criteria.pageIndex * criteria.perPage, criteria.perPage)
+        )
+      );
+    }
```
Comment (Avinkav) [inline src/app/services/words.service.ts]:
```
    let obs$ = this.changedWords;
    if (criteria.learn) obs$ = obs$.pipe(map(words => words.filter(x => !x.learned)));
    obs$.pipe(map(words => words.sort((word1: Word, word2: Word) => 
           this.sort(word1, word2, criteria.direction, criteria.sortColumn))));

    if (criteria.learn)
      obs$ = obs$.pipe(map(words => 
            words.slice(criteria.pageIndex * this.settings.maxWords, this.settings.maxWords)));
    else
      obs$ = obs$.pipe(map(words => 
            words.slice(criteria.pageIndex * criteria.perPage, criteria.perPage)));

    return obs$;
```

Comment (Avinkav) [reply src/app/services/words.service.ts]: @jimshowalter hey

Comment (jimshowalter) [reply src/app/services/words.service.ts]: I switched it to your code. It still doesn't work. Go to the learn page, enter two words, go to settings, change max words to 1, submit.

Comment (Avinkav) [reply src/app/services/words.service.ts]: edited. works fine on mine

Comment (jimshowalter) [reply src/app/services/words.service.ts]: What did you change?

By the way, the Angular extension for VSC doesn't require :Word for the sort operation.

Comment (Avinkav) [reply src/app/services/words.service.ts]: I've put stars around the bit I added, but it has no effect on the settings

**Pull request merged**